tuner/tuner/dispatch_constraints.py

```diff
@@ -217,25 +219,39 @@ def generate_solutions(
     )
     solver.add(z3.simplify(z3.And(constraints)))
     logger.debug(f"Initial constraints: {solver}")
+
+    int_type = ir.IntegerType.get_signless(64)
+
+    tuner_ctx = TunerContext(ir.Context(), logger)
```

Comment:
We shouldn't be creating a tuner context here. If it's needed, it should be passed as a function argument. This is likely to break object comparisons if we accidentally end up with two contexts at once.